data/success_stories/ahrefs-full-stack-web.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Full-Stack OCaml Web Application With React
+logo: success-stories/ahrefs.svg
+card_logo: success-stories/white/ahrefs.svg
+background: /success-stories/ahrefs-full-stack-bg.jpg
+theme: blue
+synopsis: "Ahrefs transitioned from PHP/jQuery to full-stack OCaml using Melange and React, eliminating team silos and enabling any engineer to contribute across their entire web application stack."
+url: https://ahrefs.com/
+priority: 2
+why_ocaml_reasons:
+- Type Safety
+- Unified Technology Stack
+- Team Efficiency
+- Integration with JavaScript Ecosystem
+- Shared Data Types
+- Developer Productivity
+- Code Reliability
+---
+
+## Challenge
+
+[Ahrefs](https://ahrefs.com/) is a Singapore-based SaaS company that provides SEO tools and marketing intelligence powered by big data. Since 2011, they've built their business around OCaml, using it for web crawling and data processing to serve thousands of customers worldwide. Today, they're trusted by 44% of Fortune 500 companies and operate as a lean, self-funded organization focused on efficiency.
+
+By 2017, Ahrefs had built a successful SEO tools business powered by OCaml on the backend, but they faced a bottleneck in web application development. Their frontend was built with PHP and jQuery while their data processing lived in OCaml. Every time frontend developers needed backend data, they had to coordinate with backend engineers to update the APIs.
+
+Ahrefs wanted engineers to be productive across the entire stack, but the technology divide made this unnecessarily difficult. The JavaScript tooling used in 2017 for the frontend of the web application was lacking compared to today's TypeScript ecosystem. Ahrefs had already built years of expertise in OCaml. The question became: could they extend OCaml's benefits to the frontend?
```

Comment:
I think the story is a bit different. We had a react app written in JavaScript. Then Reason and BuckleScript appeared. We consider migrating and eventually did, created a bunch of tools needed to make it work and eventually wanted to share types/join the toolchain/create universal code, etc.